src/SessionSetup/ConnectToStadiaWidget.cpp

```diff
@@ -92,25 +101,69 @@ ConnectToStadiaWidget::ConnectToStadiaWidget(QWidget* parent)
                    &ConnectToStadiaWidget::UpdateRememberInstance);
   QObject::connect(ui_->refreshButton, &QPushButton::clicked, this,
                    [this]() { emit InstanceReloadRequested(); });
+  QObject::connect(ui_->instancesFilterLineEdit, &QLineEdit::textChanged, &instance_proxy_model_,
+                   &QSortFilterProxyModel::setFilterFixedString);
+  QObject::connect(ui_->refreshButton_2, &QPushButton::clicked, this,
+                   [this]() { emit InstanceReloadRequested(); });
+  QObject::connect(ui_->comboBox, QOverload<int>::of(&QComboBox::activated), this,
+                   &ConnectToStadiaWidget::ProjectComboBoxActivated);
+  QObject::connect(ui_->allInstancesCheckBox, &QCheckBox::stateChanged, this, [this]() {
+    QSettings settings;
+    settings.setValue(kAllInstancesKey, ui_->allInstancesCheckBox->isChecked());
+  });
 
   SetupStateMachine();
 
   // The instance settings (mainly project selection and "all instances") is currently hidden behind
   // the "--enable_project_selection" flag.
   // TODO(b/190670843): Clean this up when removing the "--enable_project_selection" flag.
   if (absl::GetFlag(FLAGS_enable_project_selection)) {
-    // TODO(b/199717843): The instanceFilterLineEdit is currently hidden because it is lower
-    // priority as the rest of the UI.
-    ui_->instancesFilterLineEdit->hide();
     // While the "--enable_project_selection" is used, there are 2 refresh buttons in the .ui file.
     // "refreshButton" is used for the old ui, "refreshButton_2" is used for the new ui.
     ui_->refreshButton->hide();
+
+    ui_->comboBox->addItem("Default Project", QVariant());
+    QVariant saved_project_id = settings.value(kSelectedProjectIdKey);
+    if (saved_project_id.isValid()) {
+      const Project project{settings.value(kSelectedProjectDisplayNameKey).toString(),
+                            saved_project_id.toString()};
+      LOG("Found previously selected project. display name: %s, id: %s",
+          project.display_name.toStdString(), project.id.toStdString());
+      SetProject(project);
+      ui_->comboBox->addItem(project.display_name, project.id);
+      ui_->comboBox->setCurrentIndex(ui_->comboBox->count() - 1);
+    }
+
+    if (settings.contains(kAllInstancesKey)) {
+      ui_->allInstancesCheckBox->setChecked(settings.value(kAllInstancesKey).toBool());
+    }
+
   } else {
     // refreshButton_2 is part of instancesSettingsWidget and therefore also hidden.
     ui_->instancesSettingsWidget->hide();
   }
 }
 
+void ConnectToStadiaWidget::ProjectComboBoxActivated() {
```

Comment:
Shouldn't this technically be `ProjectComboBoxActivated(int)`?

Reply:
Using the passed index would make less assumptions about how `QComboBox` works, and it would be cleaner to read. You assume that once this signal is emitted `QComboBox::currentData` returns the same value as `QComboBox::itemData(int)` using the passed integer. This is probably reasonable, but not guaranteed from any spec as far as I can see. In theory the signal could be emitted before all of the internal states required for `currentData` would be set.

Reply:
Okay, makes sense! done